names/names.py

```diff
@@ -11,18 +12,37 @@
 f.close()
 
 duplicates = []  # Return the list of duplicates in this data structure
+# dll_names_1 = BinarySearchTree(names_1[0])
+# dll_names_2 = BinarySearchTree(names_2[0])
+
+# # Replace the nested for loops below with your improvements
+# for i in names_1:
+#     dll_names_1.insert(i)
+# for i in names_2:
+#     dll_names_2.insert(i)
+
+
+# def check_for_dupes(name):
+#     if dll_names_2.contains(name):
+#         duplicates.append(name)
+dll_names_1 = {}
+dll_names_2 = {}
+for i in range(len(names_1)):
+    dll_names_1[i] = names_1[i]
+for i in range(len(names_2)):
+    dll_names_2[names_2[i]] = i
+
+for i in range(len(names_1)):
+    if dll_names_1[i] in dll_names_2:
+        key = dll_names_1[i]
+        duplicates.append(dll_names_1[i])
```

Comment:
you can use a binary search to get this complexity down even further. At least it's O(n) though and you got it under the time constraint! Woot!